### ISSUE TITLE
Upgrade to Babel 6

### ISSUE THREAD

**Issue by ianobermiller** [FormidableLabs/radium#430]
Should solve some funky issues around Babel reading in parent `.babelrc`s as we see in #420.

**Comment by rofrischmann** [comment]
I might want to warn you about certain things I encountered when upgrading Look to Babel 6.

1. Decorators seem to be broken at all (stage-1 or even stage-0 won't help either)
2. You might need [babel-plugin-add-module-exports](https://github.com/59naga/babel-plugin-add-module-exports) in order to use export default only.
3. Somehow I can't call `babel` (cli) from npm scripts. (see [stack overflow](http://stackoverflow.com/questions/33985164/babel-doesnt-run-from-npm-script))
4. Class properties such as `onClick = () => {}` won't work without calling `super` by using a explicit constructor `constructor() {super(...arguments)}`

**Comment by epsitec** [comment]
Yes, I can confirm that `@decorators` won't work. See discussion at https://phabricator.babeljs.io/T2645 where **sebmck** wrote:

> A lot of people are subscribing to this (thank you Phabricator!) so I figure I'll give some more context and an update.
> Decorators were removed in Babel 6 because the decorator proposal has vastly changed from the version that's in Babel 5. I could implement the old proposal in Babel 6 which would make existing Babel 5 decorator code work but when I updated the transformer for the new version of the proposal, your code would break and I'd be violating semver. I'm not willing to do another major bump either as majors in a project like Babel are extremely difficult to coordinate as there's so many integrations using it.

To call `babel` from the command line, you should not install `babel-cli`.

**Comment by rofrischmann** [comment]
@epsitec Well actually I am getting this error message when trying to run a npm script that uses the babel CLI: 
```sh
The CLI has been moved into the package `babel-cli`.
$ npm install -g babel-cli
```
I actually installed the babel-cli globally, but it does not solve the problem.

**Comment by epsitec** [comment]
I am successfully using:

```json
  "scripts": {
    "compile": "babel -d lib src",
    "prepublish": "npm run compile"
  },
  "devDependencies": {
    "babel-cli": "^6.2.0",
    "babel-core": "^6.2.1",
    "babel-plugin-transform-decorators": "^6.1.18",
    "babel-plugin-transform-react-display-name": "^6.1.18",
    "babel-preset-es2015": "^6.1.18",
    "babel-preset-react": "^6.1.18",
    "babel-preset-stage-0": "^6.1.18",
    ...
}
```

and with `.babelrc` containing:

```
{
  "presets": ["stage-0", "es2015", "react"],
  "plugins": ["transform-react-display-name", "transform-decorators"]
}
```

**Comment by jedrichards** [comment]
For decorators, there is also this: https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy

**Comment by neverfox** [comment]
I'm trying to use victory, but it's radium dependency just won't play nicely in a project that's already using Babel 6. It causes a whole slew of unfulfilled dependencies in node_modules. So it seems that either Radium users (directly or indirectly) have to hold off on Babel 6 in their projects. Another solution that wouldn't require Radium making the leap would be if it could be published to npm already built, so that it didn't need to depend on babel as a primary dependency (and it could move to devDependencies).

**Comment by jedrichards** [comment]
Possibly not helpful in your case, but for the record Radium as a top-level dependency works just fine with Babel 6. My babelrc looks like this,

```json
{
  "presets": ["es2015", "react"],
  "plugins": [
    "transform-decorators-legacy"
  ]
}
```

**Comment by neverfox** [comment]
@jedrichards Yeah, I don't even get that far. `npm install` just flat out doesn't install all of the correct dependencies when I included radium/victory *and* have my own dependencies on Babel 6. Note that this is running NPM 3.x.

**Comment by ianobermiller** [comment]
Closed by #442.
